src/sidebar/util/url.js

```diff
@@ -7,19 +7,20 @@
  *   replaceURLParams('/things/:id', {id: 'foo', q: 'bar'}) =>
  *     {url: '/things/foo', unusedParams: {q: 'bar'}}
  *
+ * @template Param
  * @param {string} url
- * @param {Record<string, string>} params
- * @return {{ url: string, unusedParams: Record<string, string>}}
+ * @param {Record<string, Param>} params
+ * @return {{ url: string, unusedParams: Record<string, Param>}}
  */
 export function replaceURLParams(url, params) {
```

Comment:
The changes here make `replaceURLParams` generic so that it can be passed values which are numbers, booleans as well as strings. ie. `replaceURLParams(someURL, { someParam: 42 })` should work.

`services/api.js` was already relying on this behavior, but the mismatch between `replaceURLParams` declared types and how it was used went unnoticed due to some complete types in `services/api.js`.